### ISSUE TITLE
False positives when using identical_operands with nested array

### ISSUE THREAD

**Issue by konomae** [realm/SwiftLint#2467]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

False positives when using identical_operands with nested array.
SwiftLint 0.28.1

.swiftlint.yml

```yml
whitelist_rules:
  - identical_operands
```

main.swift

```swift
Array<Array<Int>>()
```

##### Complete output when running SwiftLint, including the stack trace and command used

```
Loading configuration from '.swiftlint.yml'
Linting Swift files at paths 
Linting 'main.swift' (1/1)
/Users/user/swiftlint-test/main.swift:1:1: warning: Identical Operands Violation: Comparing two identical operands is likely a mistake. (identical_operands)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? `0.28.1`
* Installation method used (Homebrew, CocoaPods, building from source, etc)? `cocoapods`
* Paste your configuration file:

```yml
whitelist_rules:
  - identical_operands
```

**Comment by jpsim** [comment]
Fixed in 0.29.4.
